lib/compiler/src/beam_validator.erl

```diff
@@ -967,31 +961,10 @@ vi({bs_set_position, Ctx, Pos}, Vst0) ->
 %%
 %% Floating-point instructions (excluding BIFs)
 %%
-vi({fconv,Src,{fr,_}=Dst}, Vst) ->
-    assert_term(Src, Vst),
-
-    branch(?EXCEPTION_LABEL, Vst,
-           fun(FailVst) ->
-                   %% This is a hack to supress assert_float_checked/1 in
-                   %% fork_state/2, since this instruction is legal even when
-                   %% the state is unchecked.
-                   set_fls(checked, FailVst)
-           end,
-           fun(SuccVst0) ->
-                   SuccVst = update_type(fun meet/2, number, Src, SuccVst0),
-                   set_freg(Dst, SuccVst)
-           end);
-vi(fclearerror, Vst) ->
-    case get_fls(Vst) of
-        undefined -> ok;
-        checked -> ok;
-        Fls -> error({bad_floating_point_state,Fls})
-    end,
-    set_fls(cleared, Vst);
-vi({fcheckerror,_}, Vst0) ->
-    assert_fls(cleared, Vst0),
-    Vst = set_fls(checked, Vst0),
-    branch(?EXCEPTION_LABEL, Vst);
+vi({fconv,Src,{fr,_}=Dst}, Vst0) ->
+    assert_term(Src, Vst0),
+    Vst = update_type(fun meet/2, number, Src, Vst0),
+    set_freg(Dst, Vst);
```

Comment:
Instructions that throw exceptions must branch to `?EXCEPTION_LABEL` to validate the stack and catch state. `update_type/4` also raises type conflict errors when the validator knows the types are incompatible (`none`), causing validation to fail unless used within a branch handler.

```suggestion
vi({fconv,Src,{fr,_}=Dst}, Vst) ->
    assert_term(Src, Vst),
    branch(?EXCEPTION_LABEL, Vst,
           fun(SuccVst0) ->
                   SuccVst = update_type(fun meet/2, number, Src, SuccVst0),
                   set_freg(Dst, SuccVst)
           end);
```